providers/jetstream/jsprovider.go

```diff
@@ -0,0 +1,367 @@
+package jetstream
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/nats-io/jwt/v2"
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+	"github.com/nats-io/nkeys"
+	ab "github.com/synadia-io/jwt-auth-builder.go"
+)
+
+type Provider struct {
+	natsURL    string
+	bucket     string
+	nc         *nats.Conn
+	js         jetstream.JetStream
+	kv         jetstream.KeyValue
+	encryptKey nkeys.KeyPair
+}
+
+const (
+	OperatorPrefix = "O"
+)
+
+func NewJetstreamProvider(natsURL string, bucket string, encrypt string) (ab.AuthProvider, error) {
+	p := &Provider{natsURL: natsURL, bucket: bucket}
+	if encrypt != "" {
+		kp, err := nkeys.FromCurveSeed([]byte(encrypt))
+		if err != nil {
+			return nil, err
+		}
+		p.encryptKey = kp
+	}
+	if err := p.init(); err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+func (p *Provider) init() error {
+	var err error
+	p.nc, err = nats.Connect(p.natsURL)
+	if err != nil {
+		return err
+	}
+	js, err := jetstream.New(p.nc)
+	if err != nil {
+		p.disconnect()
+		return err
+	}
+	p.js = js
+	_, err = p.js.AccountInfo(context.Background())
+	if err != nil {
+		p.disconnect()
+		return err
+	}
+	p.kv, err = p.js.KeyValue(context.Background(), p.bucket)
```

Comment:
should maye accept a context on `NewJetstreamProvider()` and get rid of all the `context.Background()`